src/catalog/src/lib.rs

```diff
@@ -151,7 +151,7 @@ pub type Epoch = NonZeroI64;
 ///
 /// If a catalog is not opened, then resources should be release via [`Self::expire`].
 #[async_trait]
-pub trait OpenableDurableCatalogState<D: DurableCatalogState>: Debug + Send {
+pub trait OpenableDurableCatalogState: Debug + Send {
```

Comment:
At some point (not this PR) it might be worth while to see if I can convert this trait into an enum